cmake/Settings.cmake

```diff
@@ -0,0 +1,34 @@
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See http://swift.org/LICENSE.txt for license information
+# See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+# Settings intended to be applied to every Swift target in this package.
+# Analogous to project-level build settings in an Xcode project.
+add_compile_definitions(
+  "$<$<COMPILE_LANGUAGE:CXX>:_SWT_TESTING_LIBRARY_VERSION=unknown (Swift 5.10 toolchain)>"
```

Comment:
`_SWT_TESTING_LIBRARY_VERSION` needs to be defined as a C++ macro that resolves to a C string containing the library version. When building with SwiftPM, we get this from `Context.gitInformation`:

```swift
    // Capture the testing library's version as a C++ string constant.
    if let git = Context.gitInformation {
      let testingLibraryVersion = if let tag = git.currentTag {
        tag
      } else if git.hasUncommittedChanges {
        "\(git.currentCommit) (modified)"
      } else {
        git.currentCommit
      }
      result.append(.define("_SWT_TESTING_LIBRARY_VERSION", to: #""\#(testingLibraryVersion)""#))
    }
```

I don't know what's possible with CMake here, but it would be ideal if we could define the equivalent when building with CMake. It would be incorrect to hard-code it to `"unknown (Swift 5.10 toolchain)"` (we do that for compatibility reasons when building with an older SwiftPM.)

Reply:
I saw this in the `Package.swift` but wasn't sure how critical it was.  My assumption was that it was mainly for debugging. I'm pretty sure I can implement this in a fairly straight-forward manner though.

Reply:
Take a look at `Package@swift-6.0.swift` instead as our canonical package manifest.

Reply:
From the LLVM repo, if it helps: https://github.com/apple/llvm-project/blob/next/llvm/cmake/modules/VersionFromVCS.cmake

Reply:
Ah, I went and found the git commands SPM executes to make sure I was matching.  Though the `QUIET` option on `find_package()` is something I didn't think of.  I should add that.